src/engine/controls/cuecontrol.cpp

```diff
@@ -567,6 +657,45 @@ void CueControl::hotcueSet(HotcueControl* pControl, double v) {
     }
 }
 
+void CueControl::hotcueSetLoop(HotcueControl* pControl, double v) {
+    //qDebug() << "CueControl::hotcueSetLoop" << v;
+
+    if (!v)
+        return;
+
+    QMutexLocker lock(&m_mutex);
+    if (!m_pLoadedTrack)
+        return;
+
+    int hotcue = pControl->getHotcueNumber();
+    // Note: the cue is just detached from the hotcue control
+    // It remains in the database for later use
+    // TODO: find a rule, that allows us to delete the cue as well
+    // https://bugs.launchpad.net/mixxx/+bug/1653276
+    hotcueClear(pControl, v);
+
+    CuePointer pCue(m_pLoadedTrack->createAndAddCue());
+    double loopStartPosition = m_pLoopStartPosition->get();
+    double loopLength = m_pLoopEndPosition->get() - loopStartPosition;
+    if (loopStartPosition < 0 || loopLength <= 0) {
+        return;
+    }
+
+    pCue->setPosition(loopStartPosition);
+    pCue->setLength(loopLength);
+    pCue->setHotCue(hotcue);
+    pCue->setLabel("");
+    pCue->setType(Cue::CUE);
+    pCue->setSource(Cue::MANUAL);
```

Comment:
will be removed in #2103 (71f2797)

Reply:
I think you can go ahead and remove this from CueControl now before #2103 is merged.

Reply:
I'd rather wait until #2103 has been merged, then pull in the changes from `master`. My codebase still expects that attribute at the moment, so removing it just in the newly introduced functions might cause strange bugs. I tried to cherry pick 107c61d270ebd6350825fa3a2a67d9ed5ed24dbb, but there are merge conflicts, so i'll either have to duplicate more of your code or just wait ;-)